radix-engine/src/transaction/toolkit_transaction_receipt/receipt/runtime.rs

```diff
@@ -0,0 +1,290 @@
+use crate::transaction::*;
+use radix_common::prelude::*;
+use radix_engine_interface::prelude::{MetadataValue, *};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct RuntimeTypeSelector;
+impl TypeSelector for RuntimeTypeSelector {
+    type Usize = usize;
+    type Bytes = Vec<u8>;
+    type Decimal = Decimal;
+
+    type NodeId = NodeId;
+    type NonFungibleGlobalId = NonFungibleGlobalId;
+
+    type MetadataValue = MetadataValue;
+
+    type WorktopChange = WorktopChange;
+}
+
+/// The runtime toolkit transaction receipt.
+pub type RuntimeToolkitTransactionReceipt = ToolkitTransactionReceipt<RuntimeTypeSelector>;
+
+impl RuntimeToolkitTransactionReceipt {
+    pub fn into_serializable_receipt(
+        self,
+        address_encoder: &AddressBech32Encoder,
+    ) -> Result<SerializableToolkitTransactionReceipt, ToolkitReceiptError> {
+        self.contextual_try_into(address_encoder)
+    }
+}
+
+impl TryFrom<VersionedTransactionReceipt> for RuntimeToolkitTransactionReceipt {
+    type Error = ToolkitReceiptError;
+
+    fn try_from(value: VersionedTransactionReceipt) -> Result<Self, Self::Error> {
+        match TransactionReceiptVersions::from(value) {
+            TransactionReceiptVersions::V1(receipt) => receipt.try_into(),
+        }
+    }
+}
+
+impl TryFrom<TransactionReceiptV1> for RuntimeToolkitTransactionReceipt {
+    type Error = ToolkitReceiptError;
+
+    fn try_from(value: TransactionReceiptV1) -> Result<Self, Self::Error> {
+        match value {
+            TransactionReceiptV1 {
+                result:
+                    TransactionResult::Commit(CommitResult {
+                        outcome: TransactionOutcome::Success(..),
+                        state_update_summary,
+                        state_updates,
+                        execution_trace: Some(execution_trace),
+                        application_events,
+                        ..
+                    }),
+                fee_summary,
+                ..
+            } => Ok(Self::CommitSuccess {
+                state_updates_summary: StateUpdatesSummary {
+                    new_entities: state_update_summary
+                        .new_components
+                        .into_iter()
+                        .map(|value| value.into_node_id())
+                        .chain(
+                            state_update_summary
+                                .new_resources
+                                .into_iter()
+                                .map(|value| value.into_node_id()),
+                        )
+                        .chain(
+                            state_update_summary
+                                .new_packages
+                                .into_iter()
+                                .map(|value| value.into_node_id()),
+                        )
+                        .collect(),
+                    // We get the metadata updates from the events.
```

Comment:
I think that either state updates or events would work here. I went with events since it simpler to get the updates this way as compared to getting them through the state updates.

Reply:
:+1: And just to check - if we create, delete, create, delete, create, is that tracked properly as a create from the events?

Reply:
I believe that this should be tracked correctly. I will add a test case for this to make sure that we track it correctly.